src/utils/pubsub/createTopic.js

```diff
@@ -0,0 +1,38 @@
+/*
+
+	ard-eventhub
+	by SWR audio lab
+
+*/
+
+// load pubsub for internal queues
+const pubSubClient = require('./_client');
+
+module.exports = async (newTopic) => {
+	let topicId = newTopic.pubsub;
+	let metadata = {
+		labels: { name: newTopic.label },
+	};
+
+	return new Promise((resolve, reject) => {
+		pubSubClient.createTopic(topicId, function (err, topic, apiResponse) {
+			if (!err) {
+				// The topic was created successfully.
+				console.log(`Topic for "${newTopic.name}" was created`);
+
+				topic.setMetadata(metadata, (err) => {
+					if (!err) {
+						// The metadata was added successfully.
+						console.log(`Metadata ${JSON.stringify(metadata)} were added`);
+					} else {
+						reject(err);
+					}
+				});
+			} else {
+				reject(err);
+			}
+			// return response
+			resolve(apiResponse);
+		});
+	});
```

Comment:
would this also work with await instead of nested promises?

Reply:
solution found using 

- publisherClient: https://github.com/googleapis/nodejs-pubsub/blob/74fc12e8d1e26939309748536bfecb751be0f079/samples/publishWithRetrySettings.js#L48
- topic-objetc: https://googleapis.dev/nodejs/pubsub/latest/google.pubsub.v1.ITopic.html

Reply:
solved with https://github.com/swrlab/ard-eventhub/pull/26/commits/92e7bd4ae4ddbb6d4f28f5b00bdc6a82b268ae59